test/unit/test/pretty_printing_test.cpp

```diff
@@ -108,3 +124,37 @@ TEST(pretty_printing, seqan3_detail_output)
     EXPECT_EQ(debug_str(seqan3::detail::my_type{"HALLO"s}), "HALLO"s);
     EXPECT_EQ(seqan3::detail::my_type{"HALLO"}, seqan3::detail::my_type{"HALLO"}); // change value to test
 }
+
+namespace seqan3
+{
+
+struct your_type : public detail::my_type
+{
+    bool operator==(your_type const & other) const
```

Comment:
aren't the comparators inherited too? (and because your_type can be implicitly casted to my_type)